meta/autoid/autoid.go

```diff
@@ -372,6 +439,96 @@ func CalcNeededBatchSize(base, n, increment, offset int64, isUnsigned bool) int6
 	return nr - base
 }
 
+// CalcSequenceBatchSize calculate the next sequence batch size.
+func CalcSequenceBatchSize(base, size, increment, offset, MIN, MAX int64) (int64, error) {
+	// The sequence is positive growth.
+	if increment > 0 {
+		if increment == 1 {
+			// Sequence is already allocated to the end.
+			if base >= MAX {
+				return 0, ErrAutoincReadFailed
```

Comment:
The error message of `ErrAutoincReadFailed` is specific to auto-increment. Perhaps another error like `ErrSeqValueReadFailed` needs to be defined.

Reply:
It will be reformed in the upper layer, see `tables.go` L1279